### PR TITLE
NTP-799: Backlink links back to specific Tp

### PR DESCRIPTION
**Pull request by Kolapo-Abolarinwa**
## Context

If a user goes into a TP detail from either the 'search results' or the 'all TPs' pages', and then uses the ‘Back’ link to go back to the list page, they are taken to the very top of the page.

## Changes proposed in this pull request

Add an id to each row in the results, then we could easily use a fragment to link the user back to that specific TP in the list.

## Guidance to review

When a user clicks the ‘Back’ button on the detail page from either the 'search results' or the 'all TPs' pages', they are taken to that TP in the list rather than the top of the page 

## Link to Jira ticket

[NTP-799](https://dfedigital.atlassian.net/browse/NTP-799)

## Things to check

- [X] Title is in the format "NTP-XXX: Short description" where NTP-XXX is the Jira ticket reference
- [ ] Code and tests follow the [coding standards](/docs/coding-standards.md)
- [X] `dotnet format` has been run in the repository root
- [ ] Test coverage of new code is at least 80%
- [ ] All UI related acceptance criteria specified in the ticket have been added to the relevant [cypress test feature file(s)](/UI/cypress/e2e/)
- [ ] Unless explicitly mentioned in the ticket, all UI work should have been tested as working on desktop and mobile resolutions with JavaScript on and off
- [ ] Database migrations can be applied to the current codebase in main without causing exceptions
- [ ] Debug logging has been added after all logic decision points
- [ ] All [automated testing](/README.md#testing) including accessibility and security has been run against your code
- [ ] **PR deployment has been signed off by wider team**

[NTP-799]: https://dfedigital.atlassian.net/browse/NTP-799?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: UI/HelperFunctions/ShortlistFunctions.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+namespace UI.HelperFunctions;
+
+public static class ShortlistFunctions
```
Comment (kieranohanlon) [inline UI/HelperFunctions/ShortlistFunctions.cs:3]: Good plan to move this in to a helper function - but not sure this is a "Shortlist" specific function?  It's used by various pages?

File: UI/cypress/e2e/full-list.js [hunk 2]
```diff
@@ -99,15 +99,14 @@ Then(
 );
 
 Then("the name of each tuition partner links to their details page", () => {
```
Comment (kieranohanlon) [inline UI/cypress/e2e/full-list.js:101]: May not be so easy to do and may have already considered it - any new cypress tests needed?

File: UI/HelperFunctions/ShortlistFunctions.cs (hunk 1 above)
Comment (Kolapo-Abolarinwa) [reply UI/HelperFunctions/ShortlistFunctions.cs:3]: You are right. The plan was to change the name but I forgot to change it - nice spot - thanks👍

File: UI/cypress/e2e/full-list.js (hunk 2 above)
Comment (Kolapo-Abolarinwa) [reply UI/cypress/e2e/full-list.js:101]: I see what I can do about this👍

**Review by kieranohanlon — APPROVED** [review]
Looks good to me

File: UI/Pages/TuitionPartner.cshtml [hunk 3]
```diff
@@ -1,263 +1,254 @@
...
     ViewData["BackLinkText"] = pageName;
     ViewData["IncludePrintPage"] = true;
 }
 <div class="govuk-grid-row">
-    <div class="govuk-grid-column-two-thirds">
-        <span show-if="@Model.Data.LocalAuthorityName != null" class="govuk-caption-l" data-testid="la-name">Tuition partner for <strong>@Model.Data.LocalAuthorityName</strong></span>
-        <h1 class="govuk-heading-l">@Model.Data.Name</h1>
-        <p class="govuk-body" data-testid="results-description">@Model.Data.Description</p>
-        <govuk-details data-testid="qatp-details">
+<div class="govuk-grid-column-two-thirds">
```
Comment (ashward) [inline UI/Pages/TuitionPartner.cshtml:15]: The indentation seems to have messed up here!

**Pull request merged**